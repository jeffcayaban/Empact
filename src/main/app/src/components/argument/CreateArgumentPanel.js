import React from "react";
import {Col, Collapse, Grid, Well} from "react-bootstrap";
import {ARGUMENT} from "../../constants";
import CreateArgumentInvite from "./create-argument/CreateArgumentInvite";

/**
 * Provides the option for the user to create an argument.
 */

class CreateArgumentPanel extends React.PureComponent {

    constructor(props) {
        super(props);

        this.state = {
            showCreateArgument: true
        }
    }

    // Updates the component's state to show or hide the create argument invitation.
    handleShowCreateArgumentInvite() {
        this.setState({ showCreateArgument: !this.state.showCreateArgument })
    }

    render() {
        const { argumentId, isRootPetitionClosed, isArgumentCreator } = this.props;

        if (!isRootPetitionClosed) {
            // Assumes the root petition is open.

            return (
                <Collapse in={this.state.showCreateArgument}>
                    <Grid style={ { paddingTop: '15px' } } className={"divNoPadding"} >
                        <Col md={12} lg={12} className={'noSidePadding'}>
                            <Well>
                                <CreateArgumentInvite
                                    isContentCreator={isArgumentCreator}
                                    contentId={argumentId}
                                    contentType={ARGUMENT}
                                    closeHandler={() => this.handleShowCreateArgumentInvite()}
                                />
                            </Well>
                        </Col>
                    </Grid>
                </Collapse>
            )
        } else {
            // Assumes the root petition is closed, of which the option to create an argument will not be provided.
            return null;
        }
    }
}

export default CreateArgumentPanel;