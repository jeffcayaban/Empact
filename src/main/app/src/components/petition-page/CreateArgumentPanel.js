import React from "react";
import {Col, Collapse, Grid} from "react-bootstrap";
import CreateArgumentInvite from "../argument/create-argument/CreateArgumentInvite";
import {PETITION} from "../../constants";
import {isDatePassed} from "../../utils/Helpers";

/**
 * Generates the container for creating a new argument. This is used for the petition page.
 */

class CreateArgumentPanel extends React.PureComponent {

    render() {
        const { closingDateTime, petitionId, isPetitionCreator, hideCreateArgumentInvite, showCreateArgument} = this.props;

        if (!isDatePassed(closingDateTime)) {
            // Assumes the root petition is still open.

            return (
                <Collapse in={showCreateArgument}>
                    <Grid className={"noSidePadding"} >
                        <Col md={12} lg={12} id={"createArgumentInvite"} className={'noSidePadding'}>
                            <CreateArgumentInvite
                                isContentCreator={isPetitionCreator}
                                contentId={petitionId}
                                contentType={PETITION}
                                closeHandler={hideCreateArgumentInvite}
                            />
                        </Col>
                    </Grid>
                </Collapse>
            )
        } else {
            // Assumes the root petition is closed, so no argument can not be created.
            return null;
        }
    }
}

export default CreateArgumentPanel;