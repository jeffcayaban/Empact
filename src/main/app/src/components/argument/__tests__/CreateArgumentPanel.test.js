import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import CreateArgumentPanel from "../CreateArgumentPanel";

describe('CreateArgumentPanel', () => {

    it('should render nothing given the petition is closed', () => {
        const component = shallow(
            <CreateArgumentPanel
                argumentId={"testArgumentId"}
                isRootPetitionClosed={true}
                isArgumentCreator={true}
            />
        );

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render successfully given the petition is not closed', () => {
        const component = shallow(
            <CreateArgumentPanel
                argumentId={"testArgumentId"}
                isRootPetitionClosed={false}
                isArgumentCreator={true}
            />
        );

        expect(toJson(component)).toMatchSnapshot();
    });


    it('should successfully call handleShowCreateArgumentInvite', () => {
        const component = shallow(<CreateArgumentPanel argumentId={"testArgumentId"} />);
        const instance = component.instance();

        const oldShowCreateArgumentValue = component.state('showCreateArgument');
        instance.handleShowCreateArgumentInvite();
        const newShowCreateArgumentValue = component.state('showCreateArgument');

        expect(newShowCreateArgumentValue).not.toBe(oldShowCreateArgumentValue);
    });
});
