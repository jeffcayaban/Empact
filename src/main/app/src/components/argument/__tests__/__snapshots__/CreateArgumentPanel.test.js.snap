// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateArgumentPanel should render nothing given the petition is closed 1`] = `""`;

exports[`CreateArgumentPanel should render successfully given the petition is not closed 1`] = `
<Collapse
  appear={false}
  dimension="height"
  getDimensionValue={[Function]}
  in={true}
  mountOnEnter={false}
  timeout={300}
  unmountOnExit={false}
>
  <Grid
    bsClass="container"
    className="divNoPadding"
    componentClass="div"
    fluid={false}
    style={
      Object {
        "paddingTop": "15px",
      }
    }
  >
    <Col
      bsClass="col"
      className="noSidePadding"
      componentClass="div"
      lg={12}
      md={12}
    >
      <Well
        bsClass="well"
      >
        <CreateArgumentInvite
          closeHandler={[Function]}
          contentId="testArgumentId"
          contentType="Argument"
          isContentCreator={true}
        />
      </Well>
    </Col>
  </Grid>
</Collapse>
`;
